### ISSUE TITLE
Implement onWorkerEnd hook

### ISSUE THREAD

**Issue by nextlevelbeard** [webdriverio/webdriverio#6356]
We've added `onWorkerStart` and toyed with adding `onWorkerEnd` before

> https://github.com/webdriverio/webdriverio/pull/4926#issuecomment-569924444

This would enable users to better manage testing dependencies or a pool of resources for the worker.

**Comment by christian-bromann** [comment]
@nextlevelbeard how would this hook be different from the `after` hook?

**Comment by Nouxx** [comment]
Hi, @christian-bromann 
There is a big difference since after hook is triggered after all tests are done.

``` 
/**
     * Gets executed after all tests are done. You still have access to all global variables from
     * the test.
     * @param {Number} result 0 - test pass, 1 - test fail
     * @param {Array.<Object>} capabilities list of capabilities details
     * @param {Array.<String>} specs List of spec file paths that ran
     */
    after: function (result, capabilities, specs) {
    }
```

There are similar hooks (AfterXXX for Cucumber let's say) but it doesn't have access to the same args as `OnWorkerStart` (in my situation described in #7237, capabilities )

**Comment by christian-bromann** [comment]
@Nouxx that might be a wrong wording in the docs then. The `after` hook is triggered after the worker finished the test. The only difference between `after` and `onWorkerEnd` is that in the `after` hook the session is still alive (`deleteSession` is not yet executed).

**Comment by Nouxx** [comment]
Well in this case that works for me
The only problem is that this `after` hook isn't triggered if an error is thrown, or maybe I'm doing it wrong.
I'll address this issue in another issue if necessary, thanks !

**Comment by christian-bromann** [comment]
This was implemented in #8100
